processor_test.go

```diff
@@ -324,3 +325,111 @@ func TestProcessor_StateReader(t *testing.T) {
 
 	require.Equal(t, ProcStateRunning, p.StateReader().State())
 }
+
+func TestProcessor_Stop(t *testing.T) {
+	t.Run("expected-state", func(t *testing.T) {
```

Comment:
If we do define subtests, then we could reduce to one top-level testing and use both `t.Run`s in side it. 

Like
```go

func TestProcessor_Stop(t*testing.T){
  t.Run("expected-state", func(t*testing.T){
   // ...
  }

  t.Run("done-closes", func(t *testing.T) {
   // ....
  })
}
```

Does that make sense?

Reply:
Yup! I went with different tests because we're technically testing different methods, but will combine them into one top level test 🙏🏼

Reply:
yeah splitting would be fine too, just having two top levels with each one sub-test looks strange. So either combine or remove the subtest.

Reply:
Ah fair, I've combined them into one function for now.